File: cpp-tiny/lib/Models/Draft.cpp


#include "Draft.h"

using namespace Tiny;

Draft::Draft()
{
	id = int(0);
	type = std::string();
	to = std::list<int>();
	topic = std::string();
	content = std::string();
	timestamp = float(0);
}

Draft::Draft(std::string jsonString)
{
	this->fromJson(jsonString);
}

Draft::~Draft()
{

}

void
Draft::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *idKey = "id";
    
    if(object.has_key(idKey)) 
    {
        bourne::json value = object[idKey];


        
        jsonToValue(&id, value, "int");


    }

    const char *typeKey = "type";
    
    if(object.has_key(typeKey)) 
    {
        bourne::json value = object[typeKey];


        
        jsonToValue(&type, value, "std::string");


    }

    const char *toKey = "to";
    
    if(object.has_key(toKey)) 
    {
        bourne::json value = object[toKey];


        std::list<int> to_list;
        int element;
        for(auto& var : value.array_range())
        {

            jsonToValue(&element, var, "int");
            
            
            to_list.push_back(element);
        }
        to = to_list;


    }

    const char *topicKey = "topic";
    
    if(object.has_key(topicKey)) 
    {
        bourne::json value = object[topicKey];


        
        jsonToValue(&topic, value, "std::string");


    }

    const char *contentKey = "content";
    
    if(object.has_key(contentKey)) 
    {
        bourne::json value = object[contentKey];


        
        jsonToValue(&content, value, "std::string");


    }

    const char *timestampKey = "timestamp";
    
    if(object.has_key(timestampKey)) 
    {
        bourne::json value = object[timestampKey];


        
        jsonToValue(&timestamp, value, "long");


    }


}

bourne::json
Draft::toJson()
{
    bourne::json object = bourne::json::object();

    



    object["id"] = getId();


    



    object["type"] = getType();


    
    
    
    std::list<int> to_list = getTo();
    bourne::json to_arr = bourne::json::array();

    for(auto& var : to_list)
    {
        to_arr.append(var);
    }
    object["to"] = to_arr;

    



    



    object["topic"] = getTopic();


    



    object["content"] = getContent();


    



    object["timestamp"] = getTimestamp();



    return object;

}

int
Draft::getId()
{
	return id;
}

void
Draft::setId(int  id)
{
	this->id = id;
}

std::string
Draft::getType()
{
	return type;
}

void
Draft::setType(std::string  type)
{
	this->type = type;
}

std::list<int>
Draft::getTo()
{
	return to;
}

void
Draft::setTo(std::list <int> to)
{
	this->to = to;
}

std::string
Draft::getTopic()
{
	return topic;
}

void
Draft::setTopic(std::string  topic)
{
	this->topic = topic;
}

std::string
Draft::getContent()
{
	return content;
}

void
Draft::setContent(std::string  content)
{
	this->content = content;
}

long
Draft::getTimestamp()
{
	return timestamp;
}

void
Draft::setTimestamp(long  timestamp)
{
	this->timestamp = timestamp;
}

File: java-vertx/src/main/java/org/openapitools/server/api/model/Draft.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

/**
 * A dictionary for representing a message draft. 
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Draft   {
  
  private Integer id;


  public enum TypeEnum {
    EMPTY(""),
    STREAM("stream"),
    PRIVATE("private");

    private String value;

    TypeEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return value;
    }
  }

  private TypeEnum type;
  private List<Integer> to = new ArrayList<>();
  private String topic;
  private String content;
  private BigDecimal timestamp;

  public Draft () {

  }

  public Draft (Integer id, TypeEnum type, List<Integer> to, String topic, String content, BigDecimal timestamp) {
    this.id = id;
    this.type = type;
    this.to = to;
    this.topic = topic;
    this.content = content;
    this.timestamp = timestamp;
  }

    
  @JsonProperty("id")
  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

    
  @JsonProperty("type")
  public TypeEnum getType() {
    return type;
  }
  public void setType(TypeEnum type) {
    this.type = type;
  }

    
  @JsonProperty("to")
  public List<Integer> getTo() {
    return to;
  }
  public void setTo(List<Integer> to) {
    this.to = to;
  }

    
  @JsonProperty("topic")
  public String getTopic() {
    return topic;
  }
  public void setTopic(String topic) {
    this.topic = topic;
  }

    
  @JsonProperty("content")
  public String getContent() {
    return content;
  }
  public void setContent(String content) {
    this.content = content;
  }

    
  @JsonProperty("timestamp")
  public BigDecimal getTimestamp() {
    return timestamp;
  }
  public void setTimestamp(BigDecimal timestamp) {
    this.timestamp = timestamp;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Draft draft = (Draft) o;
    return Objects.equals(id, draft.id) &&
        Objects.equals(type, draft.type) &&
        Objects.equals(to, draft.to) &&
        Objects.equals(topic, draft.topic) &&
        Objects.equals(content, draft.content) &&
        Objects.equals(timestamp, draft.timestamp);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, type, to, topic, content, timestamp);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Draft {\n");
    
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    to: ").append(toIndentedString(to)).append("\n");
    sb.append("    topic: ").append(toIndentedString(topic)).append("\n");
    sb.append("    content: ").append(toIndentedString(content)).append("\n");
    sb.append("    timestamp: ").append(toIndentedString(timestamp)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
